### PR TITLE
Show info about rescheduling attempts in alloc status CLI

### PR DESCRIPTION
**Pull request by preetapan**
Shows the rescheduled allocation's ID in all modes

In verbose modes, also shows attempts vs total, relative to the alloc's modify time.

### CODE REVIEW

File: api/allocations.go [hunk 1]
```diff
@@ -162,6 +162,35 @@ func (a AllocIndexSort) Swap(i, j int) {
 	a[i], a[j] = a[j], a[i]
 }
 
+// RescheduleInfo is used to calculate remaining reschedule attempts
+// according to the given time and the task groups reschedule policy
+func (a Allocation) RescheduleInfo(t time.Time) (int, int) {
+	var reschedulePolicy *ReschedulePolicy
+	for _, tg := range a.Job.TaskGroups {
+		if *tg.Name == a.TaskGroup {
```
Comment (chelseakomlo) [inline api/allocations.go]: Does this need to be dereferenced? Same for other places below which are dereferenced.

File: api/allocations.go [hunk 2]
```diff
@@ -162,6 +162,35 @@ func (a AllocIndexSort) Swap(i, j int) {
...
+			reschedulePolicy = tg.ReschedulePolicy
+		}
+	}
+	var availableAttempts int
+	var interval time.Duration
+	if reschedulePolicy != nil {
+		availableAttempts = *reschedulePolicy.Attempts
+		interval = *reschedulePolicy.Interval
+	}
+	if a.RescheduleTracker != nil && availableAttempts > 0 && interval > 0 {
```
Comment (chelseakomlo) [inline api/allocations.go]: Comment to explain this logic

File: command/alloc_status_test.go [hunk 3]
```diff
@@ -168,6 +172,32 @@ func TestAllocStatusCommand_Run(t *testing.T) {
 		t.Fatal("expected to find alloc id in output")
 	}
 	ui.OutputWriter.Reset()
+
+	// Test reschedule attempt info
```
Comment (chelseakomlo) [inline command/alloc_status_test.go]: Should this be a new test?

File: api/allocations.go [hunk 4]
```diff
@@ -162,6 +162,35 @@ func (a AllocIndexSort) Swap(i, j int) {
...
+func (a Allocation) RescheduleInfo(t time.Time) (int, int) {
+	var reschedulePolicy *ReschedulePolicy
+	for _, tg := range a.Job.TaskGroups {
+		if *tg.Name == a.TaskGroup {
+			reschedulePolicy = tg.ReschedulePolicy
+		}
+	}
+	var availableAttempts int
+	var interval time.Duration
+	if reschedulePolicy != nil {
```
Comment (chelseakomlo) [inline api/allocations.go]: Would it be the same logic to check if reschedulePolicy is nil or a.RescheduleTracker are nil, and return early instead?

File: command/alloc_status.go [hunk 5]
```diff
@@ -274,6 +274,16 @@ func formatAllocBasicInfo(alloc *api.Allocation, client *api.Client, uuidLength
...
 	}
 
+	if alloc.RescheduleTracker != nil && len(alloc.RescheduleTracker.Events) > 0 {
+		attempts, total := alloc.RescheduleInfo(time.Unix(0, alloc.ModifyTime))
+		reschedInfo := fmt.Sprintf("Reschedule Attempts|%d/%d", attempts, total)
+		basic = append(basic, reschedInfo)
+	}
+	if alloc.NextAllocation != "" {
+		basic = append(basic,
+			fmt.Sprintf("Rescheduled Alloc ID|%s", limit(alloc.NextAllocation, uuidLength)))
```
Comment (dadgar) [inline command/alloc_status.go]: Replacement Alloc

File: api/allocations.go [hunk 6]
```diff
@@ -162,6 +162,35 @@ func (a AllocIndexSort) Swap(i, j int) {
...
+		availableAttempts = *reschedulePolicy.Attempts
```
Comment (dadgar) [inline api/allocations.go]: Why not return 0, 0 in here?

Comment (preetapan) [reply api/allocations.go]: Tried to make it work when one or the other is not set but seems like that's causing more confusion? I can change this to use early returns if that's simpler/

File: api/allocations.go (hunk 1 above)
Comment (dadgar) [reply api/allocations.go]: Yeah the allocation is raw string the job is has pointers

**Pull request merged**